### DISCUSSION TITLE
Zone crossing

### DISCUSSION THREAD

**Discussion by Yonny24** [blakeblackshear/frigate#7205] [Ideas]
Hi

Do you have any plans/thoughts of implenting zone crossings?
For example if an object is tracked through zone 1 and enters zone 2 it only creates an alert/event if it crosses. 1>2 (1 to 2 only) or 1-2(both ways)

This is very uesful on Blueiris and I use it in almost every camera set up I have. 
Thanks

**Comment by NickM-27** [comment]
Can you explain a usecase for this?

**Reply by Yonny24** [reply]
Yep e.g a motorcycle or person walks from the road onto the driveway outside my house but not the other way. (delivery for example)
Wife coming home driving in, only needs to alert me when the car passes from zone 1 to 2. I don't need alerts for a car all the time in 1 zone when it is not crossing either.
Works well in BI
During summer I would have various zones over where a large tree is casting moving shadows.
This would reduce false positives greatly.

Thanks

**Reply by NickM-27** [reply]
All of these things can be done in frigate without needing what you are describing. 

> e.g a motorcycle or person walks from the road onto the driveway outside my house but not the other way. (delivery for example)

Just make driveway a required zone and then can create a notification only in certain circumstances 

> Wife coming home driving in, only needs to alert me when the car passes from zone 1 to 2. I don't need alerts for a car all the time in 1 zone when it is not crossing either.

Right, frigate already offers full customization of notifications in this way

> During summer I would have various zones over where a large tree is casting moving shadows.
This would reduce false positives greatly.

In frigate zones are not used for this, motion masks are

**Comment by Yonny24** [comment]
Not sure where I was going wrong the other day.

I created 2 zones
Frontyard -  person,cat,dog
Outside Drive - person
Close to each other - just a high fence in between the zones and a gate where I walk threw

Had issues getting alerts/events when I was walking from Frontyard to the Outisde Drive.
I could see myself in debug but no events being created in the 2nd zone so gave up and removed the zones.
Is it because it was detecting myself as person as one long trigger through both zones?

Spent all day trying to get them to work properly.

**Reply by NickM-27** [reply]
You might be misunderstanding how zones work, an objects presence in a zone is based on the bottom center pixel of its bounding box. To be able to help specifically would need a picture of the debug live view and the config

**Comment by philber1155** [comment]
Hi, NickM, can you please elaborate about how frigate can send an alert only on a specific sequence oz zones being crossed? I created zones, I selected the ones to be considered for alerts, but I dont see where to say "alert if crossing Z1, then Z2". Is it a specific code to add in the yamL? 
Thanks

**Reply by NickM-27** [reply]
you don't specify it in the Frigate config, you specify it when you are reading the MQTT topic before sending a notification.

**Reply by philber1155** [reply]
Do you mean, you get a first mqqtt frame indicating zone 1 crossed, this starts an automation waiting for a seconf mqtt that would come indicating zone 2 crossed, and there you say  its an alert?

**Reply by NickM-27** [reply]
No, every message has a `before` and `after`. You would just check that the before.entered_zones == ["zone_1"] and after.current_zones == ["zone_2"]

**Comment by cchewanik** [comment]
Building on this idea, is there a way to specify a minimum distance (ex. in pixels) an object has moved before a snapshot or review is triggered? I see you have auto-tracking data being provided to cameras, can we tap into that feed some how through MQTT or API? 

Despite my attempts with Frigate+ I still have the occasional (got to weekly now) false detection. Just trying to kill false detections once and for all!!

You guys are awesome thanks for all your efforts!!

**Reply by NickM-27** [reply]
motion has to be detected and the object must have made a position change before it has a review item created. If this is occurring then I don't think that is the right avenue to reduce the false positives. That would likely be a separate question that we can look into

**Reply by cchewanik** [reply]
Thanks for the quick reply. The specific example I have is of an outdoor light fixture that got a snapshot and a review. In the object lifecycle, it shows a person being detected, then going stationary.

It was windy last night and snowy so that’s what probably triggered the motion. It’s a fixed object, if I could start looking at its positional data once falsely detected as a person, if it has moved x then I would trigger a notification. 

How big of a position change are you looking for now when a review is triggered?

**Reply by NickM-27** [reply]
it depends how much of the bounding box overlaps between frames. but like I said before "movement" really isn't a good way to filter out objects, there are likely better ways. Happy to make suggestions if you create a new support issue.
